classes/autoptimizeScripts.php

```diff
@@ -342,9 +342,14 @@ public function read( $options )
                             $consider_minified_array = apply_filters( 'autoptimize_filter_js_consider_minified', false );
                             if ( ( false === $this->aggregate && str_replace( $this->dontmove, '', $path ) === $path ) || ( true === $this->aggregate && ( false === $consider_minified_array || str_replace( $consider_minified_array, '', $path ) === $path ) ) ) {
                                 $minified_url = $this->minify_single( $path );
-                                // replace orig URL with minified URL from cache if so.
                                 if ( ! empty( $minified_url ) ) {
+                                    // Replace original URL with minified URL from cache.
                                     $new_tag = str_replace( $url, $minified_url, $new_tag );
+                                } else {
+                                    if ( apply_filters( 'autoptimize_filter_js_remove_empty_files', false ) ) {
```

Comment:
Wouldn't it be better to remove the filter and set `$new_tag` to `''` unconditionally here and to use the filter in line https://github.com/sjoerdlinders/autoptimize/blob/patch-4/classes/autoptimizeScripts.php#L366 the same way you did for CSS (so $new_tag is not empty and diff. from $orig_tag OR $new_tag is empty and filter is true)?

Reply:
Moved filter to correct place, for consistency between files.